tools/clang/lib/Headers/hlsl/dx/linalg.h

```diff
@@ -43,14 +43,24 @@ enum MatrixLayout {
 // Helper for signedness
 //
 namespace details {
-template <typename T> bool IsUnsigned() { return false; }
 
-#ifdef __HLSL_ENABLE_16_BIT
-template <> bool IsUnsigned<uint16_t>() { return true; }
-#endif
+template <typename T> struct IsUnsigned {
+  static const bool value = false;
```

Comment:
```suggestion
  static const bool Value = false;
```

Reply:
Phew, no line number change.. the tests needn't be updated!